test_scripts/RC/AUDIO_LIGHT_HMI_SETTINGS/commonRCmodules.lua

```diff
@@ -0,0 +1,627 @@
+---------------------------------------------------------------------------------------------------
+-- RC common module for AUDIO, LIGHT, HMI_SETTINGS modules
+---------------------------------------------------------------------------------------------------
+config.application1.registerAppInterfaceParams.appHMIType = { "REMOTE_CONTROL" }
+config.application2.registerAppInterfaceParams.appHMIType = { "REMOTE_CONTROL" }
+config.application1.registerAppInterfaceParams.syncMsgVersion.majorVersion = 5
+config.application2.registerAppInterfaceParams.syncMsgVersion.majorVersion = 5
+
+--[[ Required Shared libraries ]]
+local commonRC = require("test_scripts/RC/commonRC")
```

Comment:
Seems that the regular `commonRC` module could just be extended, rather than include a separate common module for these tests.

Reply:
no, `commonRC` uses another approach with self, and in our common we used an approach without self and also redefine some functions from initial commonRC.

Reply:
At the very least, the `rcRPCs` struct is redundant here, you should be able to just make additions to the `commonRC.lua` implementation of this code rather than implementing it entirely again in the common module. Then, just inherit the implementation of the functions that use this struct into `commonRCmodules.lua`:
```
c.getAppEventName = commonRC.getAppEventName
c.getHMIEventName = commonRC.getHMIEventName
c.getAppRequestParams = commonRC.getAppRequestParams
c.getAppResponseParams = commonRC.getAppResponseParams
c.getHMIRequestParams = commonRC.getHMIRequestParams
c.getHMIResponseParams = commonRC.getHMIResponseParams
```

Reply:
rcRPCs local table had specific changes in commonRCmodules.lua related to new modules, so can not be reused from commonRC.lua

Reply:
I was requesting that `rcRPCs` be extended in `commonRC.lua` so that this module could use it instead.

For clarity's sake, here is what I want done with this module:
1. Any functions which are duplicated from `commonRC.lua` should be just moved to `commonRC.lua` as extensions to the existing functions. For example, new parameters and modules should be added to the `commonRC.lua` implementation of `getModuleControlData`, rather than including a new implementation of the function in `commonRCmodules.lua`.
2. While it should be avoided in general, if a function cannot be extended in this way due to some aspect of it being entirely exclusive to this feature (`PTUfunc` may be an example of this, but I'm not certain right now), `commonRCmodules.lua` can be used to extend this function. This new function should call the original implementation of the function, rather than rewriting the function entirely.
3. Some specific functions/structures which seem like they could easily be extended instead of rewritten: `getModuleControlData`, `getAnotherModuleControlData`, `getModuleParams`, `getReadOnlyParamsByModule`, `rcRPCs`, `subscribeToModule`, `unSubscribeToModule`, etc.